.github/workflows/deploy.yml

```diff
@@ -62,9 +63,7 @@ jobs:
           cache-dependency-path: "pyproject.toml"
 
       - name: Install dependencies
-        run: |
-          python -m pip install --upgrade pip
```

Comment:
Have removed the upgrade `pip` step throughout (see https://github.com/actions/setup-python/issues/506#issuecomment-1262522726)

Reply:
is it worth keeping the upgrade command here just in case the version on the runner is out of date? Maybe not necessary though

Reply:
I've added back in for the `verify-testpypi-contains-binary` step